### PR TITLE
TC15 - User can open "Book ticket" page by click on "Book ticket" link in "Ticket price"

### PR DESCRIPTION
**Pull request by sattdn21-01-02**

### CODE REVIEW

File: src/main/java/page_objects/TicketPricePage.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package page_objects;
+
+import helper.BrowserHelper;
+import org.openqa.selenium.By;
+
+public class TicketPricePage extends GeneralPage {
+
+    //Elements
+    private final String dynamicBtnCheckPriceXpath = "//li[text() = '%s to %s']//ancestor::tr//a";
```
Comment (tanmle) [inline src/main/java/page_objects/TicketPricePage.java:9]: not a good xpath

File: src/main/java/page_objects/TicketPricePage.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    private final String dynamicBtnBookTicketXpath = "//td[text()='%s']//following-sibling::td/a";
```
Comment (tanmle) [inline src/main/java/page_objects/TicketPricePage.java:10]: not a good xpath

**Review by tanmle — CHANGES_REQUESTED** [review]

File: src/main/java/page_objects/TicketPricePage.java (hunk 1 above)
Comment (sattdn21-01-02) [reply src/main/java/page_objects/TicketPricePage.java:9]: thank you for the review, I will find better XPath

File: src/test/java/com/logigear/BookTicketTest.java [hunk 3]
```diff
@@ -43,4 +45,32 @@ public void TC14() {
...
+
+        homePage.goToRegisterPage();
+        registerPage.register(account);
+        homePage.goToLoginPage();
+        loginPage.login(account);
+        homePage.goToTicketPricePage();
+        ticketPricePage.clickOnCheckPrice(departFrom, arriveAt);
+        ticketPricePage.clickOnBookTicket(seatType);
+
+        BrowserHelper.scrollPage();
```
Comment (tanmle) [inline src/test/java/com/logigear/BookTicketTest.java:66]: why do we need scrollPage?

File: src/test/java/com/logigear/BookTicketTest.java [hunk 4]
```diff
@@ -43,4 +45,32 @@ public void TC14() {
...
+
+        String actualDepartFromMsg = bookTicketPage.getDepartFromValue();
```
Comment (tanmle) [inline src/test/java/com/logigear/BookTicketTest.java:68]: why do you get value but your variable is Msg?

File: src/test/java/com/logigear/BookTicketTest.java (hunk 3 above)
Comment (sattdn21-01-02) [reply src/test/java/com/logigear/BookTicketTest.java:66]: in my TC, the book ticket page not auto-scroll so I need a scrolling page to check the information book ticket. and assert actual with expected result

File: src/test/java/com/logigear/BookTicketTest.java (hunk 4 above)
Comment (sattdn21-01-02) [reply src/test/java/com/logigear/BookTicketTest.java:68]: thank you for the review, I think I should check and rename variable

File: src/test/java/com/logigear/BookTicketTest.java [hunk 5]
```diff
@@ -43,4 +45,30 @@ public void TC14() {
...
+        ticketPricePage.clickOnCheckPrice(departFrom, arriveAt);
+        ticketPricePage.clickOnBookTicket(seatType);
+
+        String actualDepartFrom = bookTicketPage.getDepartFromValue();
+        String actualArriveAt = bookTicketPage.getArriveAtValue();
+        String actualSeatType = bookTicketPage.getSeatTypeValue();
+
+        Assert.assertEquals(actualDepartFrom, departFrom, "Depart from error to display incorrect!");
+        Assert.assertEquals(actualArriveAt, arriveAt, "Arrive at error to display incorrect!");
+        Assert.assertEquals(actualSeatType, seatType, "Seat type error to display incorrect!");
```
Comment (LoliSlayerMC) [inline src/test/java/com/logigear/BookTicketTest.java:72]: error is a noun, please check grammar for above messages

**Review by LoliSlayerMC — CHANGES_REQUESTED** [review]

**Review by LoliSlayerMC — APPROVED** [review]
LGTM

**Pull request merged**